test/ShallowWrapper-spec.js

```diff
@@ -158,6 +158,19 @@ describe('shallow', () => {
       expect(wrapper.contains(passes2)).to.equal(true);
     });
 
+    it('shouldn\'t confuse non React elements and string literals in DOM', () => {
+      const functionalComponent = () => <div></div>;
+
+      const wrapper = shallow(
+        <div>
+          <functionalComponent />
+          { 'Text node' }
+        </div>
+      );
+
+      expect(wrapper.contains(functionalComponent)).to.equal(false);
```

Comment:
@ljharb That was what I thought at first, but in this case `functionalComponent` is not a ReactElement (which would be a valid argument for `contains`) but a `render` function. 

Unfortunately it is not possible for now to use `selector` as `contains` argument so this test could pass (I dont have time now to confirm that, but AFAIR it should) if we make such assertion - `expect(wrapper.contains(<functionalComponent />)).to.equal(false)` .

But due to presence of that text node the function would return (before my PR) `true`, although wouldn't return `true` for just `shallow(<div><functionalComponent></div>)`. Therefore it's outcome was kinda unpredictable and cost me some time as I didn't read the docs carefully and thought that using `selector` here is OK (still - I would love to be able to use selectors here too ;) )

@lelandrichardson 
True that, gonna amend this commit in a minute